java-frontend/src/main/java/org/sonar/plugins/java/api/JavaVisitor.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.java.api;
+
+/**
+ * Root of all visitor classes from SonarJava analyzers
```

Comment:
I would get the javadoc consistent with the one on JavaCheck : https://github.com/SonarSource/sonar-java/pull/1918/files#diff-7c14b6cca5558b9a05292852b3d9b6a2L25 

I am also wondering about the relevance of JavaCheck interface with the introduction of JavaVisitor.  (and maybe those two classes should be renamed to SonarJavaCheck (given it is also used for pom and xml) and SonarJavaVisitor ?

Reply:
I don't want to rename the class, as it is part of the API. However, I'm going to drop JavaVisitor. Xml and Maven checks are already extending JavaCheck. Issue filters as well. No need to add a new layer.